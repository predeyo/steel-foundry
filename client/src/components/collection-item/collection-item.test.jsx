import React from "react";
import { shallow } from "enzyme";
import { CollectionItem } from "./collection-item.component.jsx";

describe("CollectionItem component", () => {
  let wrapper;
  let mockAddItem;
  let mockName = "Red hat";
  let mockPrice = "99";
  let mockImageUrl = "www.testimage.com";
  beforeEach(() => {
    mockAddItem = jest.fn();
    let mockProps = {
      item: {
        name: mockName,
        price: mockPrice,
        imageUrl: mockImageUrl
      },
      addItem: mockAddItem
    };
    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it("should render CollectionItem compoenent", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should call addItem when "add to cart" button in clicked', () => {
    wrapper.find("CustomButton").simulate("click");
    expect(mockAddItem).toHaveBeenCalled();
  });

  it("should render name prop in footer", () => {
    expect(wrapper.find(".name").text()).toEqual(mockName);
  });

  it("should render price prop in footer", () => {
    expect(wrapper.find(".price").text()).toEqual(mockPrice);
  });

  it("should pass imageUrl in image container", () => {
    expect(wrapper.find(".image").get(0).props.style).toHaveProperty(
      "backgroundImage",
      `url(${mockImageUrl})`
    );
  });
});
