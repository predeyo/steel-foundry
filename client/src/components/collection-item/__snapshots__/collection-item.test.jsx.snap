// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should render CollectionItem compoenent 1`] = `
"<div className=\\"collection-item\\">
  <div className=\\"image\\" style={{...}} />
  <div className=\\"collection-footer\\">
    <span className=\\"name\\">
      Red hat
    </span>
    <span className=\\"price\\">
      99
    </span>
  </div>
  <CustomButton onClick={[Function: onClick]} inverted={true}>
    Add to cart
  </CustomButton>
</div>"
`;
